### DISCUSSION TITLE
Cross Database Populate

### DISCUSSION THREAD

**Discussion by olawalejuwonm** [typegoose/typegoose#938] [Q&A]
Hi

Please how do i specify ref as described here: https://mongoosejs.com/docs/populate.html#cross-db-populate for cross database population

**Comment by hasezoey** [comment]
you can always use the second way defined there: pass the Model parameter to the `.populate` call:

```ts
const events = await Event.
  find().
  // The `model` option specifies the model to use for populating.
  populate({ path: 'conversation', model: Conversation });
```

aside from that, you can also pass the final generated model to the class definition, but if those classes are static / module-global context, then it is not really too-convenient to use, though deferred functions are supported for `ref` (like `ref: () => Model` or pass it a function that returns the model from some place elsewhere)

**Reply by olawalejuwonm** [reply]
@hasezoey i want this defined in the schema so that I can do .populate at any time

**Reply by hasezoey** [reply]
well like i wrote in the original response, you can use deferred functions to set the model:
```ts
class T {
  @prop({ ref: () => YourOtherModel })
  public somepath: Ref<YourOtherClass>;
}
```

though note that this requires that the `YourOtherModel` gets *compiled* before that new model, so only setting it via `.populate` is a viable option if you dont heavily control the import / compile order (to my knowledge, this applies to plain mongoose too).
Alternatively, you could also use a function that does the population and is defined where your schema is.
